config/database_test.go

```diff
@@ -133,37 +133,79 @@ func TestDatabaseStoreNew(t *testing.T) {
 	sqlSettings := mainHelper.GetSQLSettings()
 
 	t.Run("no existing configuration - initialization required", func(t *testing.T) {
-		ds, err := newTestDatabaseStore(t)
+		ds, err := newTestDatabaseStore(t, nil)
 		require.NoError(t, err)
 		defer ds.Close()
 
 		assert.Equal(t, "", *ds.Get().ServiceSettings.SiteURL)
 	})
 
+	t.Run("no existing configuration with custom defaults", func(t *testing.T) {
+		truncateTables(t)
+		ds, err := newTestDatabaseStore(t, customDefaults)
+		require.NoError(t, err)
+		defer ds.Close()
+
+		assert.Equal(t, *customDefaults.ServiceSettings.SiteURL, *ds.Get().ServiceSettings.SiteURL)
+		assert.Equal(t, *customDefaults.DisplaySettings.ExperimentalTimezone, *ds.Get().DisplaySettings.ExperimentalTimezone)
+	})
+
 	t.Run("existing config, initialization required", func(t *testing.T) {
 		_, tearDown := setupConfigDatabase(t, testConfig, nil)
 		defer tearDown()
 
-		ds, err := newTestDatabaseStore(t)
+		ds, err := newTestDatabaseStore(t, nil)
 		require.NoError(t, err)
 		defer ds.Close()
 
 		assert.Equal(t, "http://TestStoreNew", *ds.Get().ServiceSettings.SiteURL)
 		assertDatabaseNotEqualsConfig(t, testConfig)
 	})
 
+	t.Run("existing config with custom defaults, initialization required", func(t *testing.T) {
+		_, tearDown := setupConfigDatabase(t, testConfig, nil)
+		defer tearDown()
+
+		ds, err := newTestDatabaseStore(t, customDefaults)
+		require.NoError(t, err)
+		defer ds.Close()
+
+		// already existing value should not be overwritten by the
+		// custom default value
+		assert.Equal(t, "http://TestStoreNew", *ds.Get().ServiceSettings.SiteURL)
+		// not existing value should be overwritten by the custom
+		// default value
+		assert.Equal(t, *customDefaults.DisplaySettings.ExperimentalTimezone, *ds.Get().DisplaySettings.ExperimentalTimezone)
+		assertDatabaseNotEqualsConfig(t, testConfig)
+	})
+
 	t.Run("already minimally configured", func(t *testing.T) {
 		_, tearDown := setupConfigDatabase(t, minimalConfig, nil)
 		defer tearDown()
 
-		ds, err := newTestDatabaseStore(t)
+		ds, err := newTestDatabaseStore(t, nil)
 		require.NoError(t, err)
 		defer ds.Close()
 
 		assert.Equal(t, "http://minimal", *ds.Get().ServiceSettings.SiteURL)
 		assertDatabaseEqualsConfig(t, minimalConfig)
 	})
 
+	t.Run("already minimally configured with custom defaults", func(t *testing.T) {
+		_, tearDown := setupConfigDatabase(t, minimalConfig, nil)
+		defer tearDown()
+
+		ds, err := newTestDatabaseStore(t, customDefaults)
+		require.NoError(t, err)
+		defer ds.Close()
+
+		// as the whole config has default values already, custom
```

Comment:
I don't understand this test. Looks like the same than line 165

Reply:
The difference is in the `setupConfigDatabase` call. In the test at line 165 is initialising the configuration with `testConfig`, which only has the `SiteURL` set, so the defaults should overwrite `ExperimentalTimezone` but not `SiteURL`, as is loaded with the config.

In the test at 202, the `setupConfigDatabase` fn is called with `minimalConfig`, which instead of a partial config is a complete one, so it has all the values set.

Reply:
Oh ok, I missed the Equal vs NotEqual